tests/test_conjugates.py

```diff
@@ -157,3 +158,49 @@ def test_beta_negative_binomial_conjugate_expand():
     expanded = eval_if_etuple(expanded_expr)
 
     assert isinstance(expanded.owner.op, type(at.random.beta))
+
+
+def test_uniform_pareto_conjugate_contract():
+    """Produce the closed-form posterior for the uniform observation model with
+    a pareto prior.
+
+    """
+    srng = RandomStream(0)
+
+    xm_tt = at.scalar("xm")
+    k_tt = at.scalar("k")
+    theta_rv = srng.pareto(k_tt, xm_tt, name="theta")
+
+    # zero = at.iscalar("zero")
+    Y_rv = srng.uniform(0, theta_rv)
+    y_vv = Y_rv.clone()
+    y_vv.tag.name = "y"
+
+    q_lv = var()
+    (posterior_expr,) = run(1, q_lv, uniform_pareto_conjugateo(y_vv, Y_rv, q_lv))
+    posterior = eval_if_etuple(posterior_expr)
+
+    assert isinstance(posterior.owner.op, type(at.random.pareto))
+
+    # Build the sampling function and check the results on limiting cases.
+    sample_fn = aesara.function((xm_tt, k_tt, y_vv), posterior)
+    assert sample_fn(1.0, 1000, 1) == pytest.approx(1.0, abs=0.01)  # k = 1000
+    assert sample_fn(1.0, 1, 0) == pytest.approx(0.0, abs=0.01)  # all zeros
+
+
+def test_uniform_pareto_binomial_conjugate_expand():
+    """Expand a contracted beta-binomial observation model."""
+
+    srng = RandomStream(0)
+
+    k_tt = at.scalar("k")
+    y_vv = at.iscalar("y")
+    n_tt = at.scalar("n")
+
+    Y_rv = srng.pareto(at.max(y_vv), k_tt + n_tt)
```

Comment:
Currently, the test graph looks like this:
```python
>>> aesara.dprint(Y_rv)
pareto_rv{0, (0, 0), floatX, False}.1 [id A]
 |RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7FC4162BFF20>) [id B]
 |TensorConstant{[]} [id C]
 |TensorConstant{11} [id D]
 |MaxAndArgmax{axis=()}.0 [id E] 'max'
 | |y [id F]
 |Elemwise{add,no_inplace} [id G]
   |k [id H]
   |n [id I]
```
and that `MaxAndArgmax` `Op` isn't same as the `at.math.max` used [in the `etuple` graph](https://github.com/aesara-devs/aemcmc/pull/113/files#diff-8ad96d062cb6bdb9af4a61717241f6f20157d2ddac6de7c37d0e0ef5e7539348R306).  `at.math.max` is a function that constructs a `MaxAndArgmax` `Op` and uses it to further construct a graph for the max of its argument.  In other words, we need an `etuple` form/"pattern" that matches the types of graphs output by the helper function `at.math.max`.

Often the easiest way to find `etuple` forms for the graphs constructed by helper functions is to `etuplize` said graphs and spot their generalities.
For example:
```python
>>> from etuples import etuplize
>>> etuplize(at.math.max(at.vector("x")))
e(e(aesara.tensor.math.MaxAndArgmax, (0,)), x)
>>> etuplize(at.math.max(at.matrix("x")))
e(e(aesara.tensor.math.MaxAndArgmax, (0, 1)), x)
```

As we can see, the axis property in the `MaxAndArgmax` `Op` will change according to the dimensions of the input (i.e. it computes the max across _all_ dimensions), so we don't want to use a very specific value for the matching form.  Instead, we can use another logic variable in place of those values.

Here's a general testing setup for that part of the problem:
```python
import aesara
import aesara.tensor as at

from etuples import etuplize, etuple


srng = at.random.RandomStream(0)

k_tt = at.scalar("k")
y_vv = at.iscalar("y")
n_tt = at.scalar("n")

Y_rv = srng.pareto(at.max(y_vv), k_tt + n_tt)

# This is what we need to match/unify:
etuplize(Y_rv)
# e(
#     e(aesara.tensor.random.basic.ParetoRV, 'pareto', 0, (0, 0), 'floatX', False),
#     RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7FA1F9B3D9E0>),
#     TensorConstant{[]},
#     TensorConstant{11},
#     e(e(aesara.tensor.math.MaxAndArgmax, ()), y),
#     e(
#         e(
#             aesara.tensor.elemwise.Elemwise,
#             <aesara.scalar.basic.Add at 0x7fa1fd3823d0>,
#             <frozendict {}>),
#         k,
#         n))

from unification import var
from kanren import run, eq
from aesara.tensor.math import MaxAndArgmax


observed_val = var()
axis_lv = var()
new_x_et = etuple(etuple(MaxAndArgmax, axis_lv), observed_val)

k_lv, n_lv = var(), var()
new_k_et = etuple(etuplize(at.add), k_lv, n_lv)

theta_rng_lv = var()
theta_size_lv = var()
theta_type_idx_lv = var()
theta_posterior_et = etuple(
    etuplize(at.random.pareto),
    theta_rng_lv,
    theta_size_lv,
    theta_type_idx_lv,
    new_x_et,
    new_k_et,
)


run(0, (new_x_et, new_k_et), eq(Y_rv, theta_posterior_et))
# ((e(e(aesara.tensor.math.MaxAndArgmax, ()), y),
#   e(
#       e(
#           aesara.tensor.elemwise.Elemwise,
#           <aesara.scalar.basic.Add at 0x7fa1fd3823d0>,
#           <frozendict {}>),
#       k,
#       n)),)
```